### PR TITLE
Allow for non-uniform PixelBoxes + Refactoring

### PR DESCRIPTION
**Pull request by Hirevo**
This PR is part of an initiative to refactor how rendering is handled originating from Corentin Leroy.
This allows non-uniformly colored PixelBoxes to be represented and passed around between libraries.

### CODE REVIEW

File: PixelBox.hpp [hunk 1]
```diff
@@ -16,24 +16,22 @@ namespace Arcade {
 	public:
 		PixelBox(size_t height, size_t width, size_t posW,
 			size_t posH);
-		size_t getH() const;
-		size_t getPosH() const;
-		void setH(size_t height);
-		void setPosH(size_t posH);
+		size_t &getHeight() const;
```
Comment (thibautcornolti) [inline PixelBox.hpp]: You can't mix `const` and `&`

File: PixelBox.hpp [hunk 2]
```diff
@@ -16,24 +16,22 @@ namespace Arcade {
...
+		size_t &getY() const;
+		void setHeight(size_t height);
+		void setY(size_t y);
 
-		size_t getW() const;
-		size_t getPosW() const;
-		void setW(size_t width);
-		void setPosW(size_t posW);
+		size_t &getWidth() const;
+		size_t &getX() const;
```
Comment (rectoria) [inline PixelBox.hpp]: cf @thibautcornolti

File: PixelBox.hpp [hunk 3]
```diff
@@ -16,24 +16,22 @@ namespace Arcade {
...
+		size_t &getHeight() const;
+		size_t &getY() const;
+		void setHeight(size_t height);
+		void setY(size_t y);
 
-		size_t getW() const;
-		size_t getPosW() const;
-		void setW(size_t width);
-		void setPosW(size_t posW);
+		size_t &getWidth() const;
```
Comment (rectoria) [inline PixelBox.hpp]: cf @thibautcornolti

File: PixelBox.hpp (hunk 1 above)
Comment (Hirevo) [reply PixelBox.hpp]: Correct, I fixed it. It's quite surprising given that standalone const references are allowed.

**Pull request merged**